src/bin/build_test.rs

```diff
@@ -0,0 +1,264 @@
+extern crate chrono;
+
+use std::collections::HashMap;
+use std::fmt::Debug;
+use std::fs::{self, File};
+use std::io::{self, Write};
+use std::path::{Path, PathBuf};
+
+use chrono::Utc;
+use syn::{self, visit::Visit, ExprArray, ItemConst, ItemStatic};
+
+struct SliceVisitor {
+    // This tracks information during the AST traversal.
+    // One of the reasons we need `inside_item` is because we define the trait
+    // method `visit_expr_array` - which means it will run unconditionally for
+    // all `ExprArray`s of a target file. But we only want to execute the code
+    // of our defined `visit_expr_array` for the ones inside `ItemStatic`s.
+    // There's more info in the comments of `visit_expr_array`
+    //
+
+    // Stores:
+    // 0. Whether the Visitor is inside an ItemStatic (not necessarily
+    // directly, it could be in an Expr, but 'true' would mean the Expr is
+    // inside an ItemStatic)
+    // 1. The Ident (name) of the Item the Visitor is in. This is used in
+    // accesses to the `slices` field
+    inside_item: (bool, String),
+
+    // Keeps track of which file is currently being parsed
+    file_name: &'static str,
+
+    // Stores tuples to track all static slices (idents) and their lengths
+    slices: HashMap<String, usize>,
+}
+
+impl<'ast> Visit<'ast> for SliceVisitor {
+    fn visit_expr_array(&mut self, expr: &'ast ExprArray) {
+        // We only do anything if we arrived here through `visit_item_check`
+        if self.inside_item.0 {
+            self.slices
+                .insert(self.inside_item.1.clone(), expr.elems.len());
+            println!("Length of {}: {}", self.inside_item.1, expr.elems.len());
+        }
+    }
+
+    fn visit_item_const(&mut self, item: &'ast ItemConst) {
+        println!("Found ItemConst named {:?}\n", item.ident);
+
+        if let syn::Expr::Reference(expr_ref) = &*item.expr {
+            if let syn::Expr::Array(expr_array) = &*expr_ref.expr {
+                self.inside_item = (true, item.ident.to_string());
+                self.visit_expr_array(expr_array);
+                self.inside_item = (false, String::from(""));
+            }
+        }
+    }
+
+    fn visit_item_static(&mut self, item: &'ast ItemStatic) {
+        println!("Found ItemStatic named {:?}\n", item.ident);
+
+        if let syn::Expr::Reference(expr_ref) = &*item.expr {
+            if let syn::Expr::Array(expr_array) = &*expr_ref.expr {
+                self.inside_item = (true, item.ident.to_string());
+                self.visit_expr_array(expr_array);
+                self.inside_item = (false, String::from(""));
+            }
+        }
+    }
+}
+
+fn main() -> io::Result<()> {
+    //
+
+    let files = {
+        vec![
+            "/home/condekind/repos/ctci-rs/src/ch01_arrays_and_strings/iq01.rs",
```

Comment:
All hardcoded paths will be replaced by relative ones once the contents of this file are moved to `build.rs`.

I created this as an individual binary to facilitate debugging, since RustRover's debugging is not straightforward when it comes to `build.rs`